cpp/cppsdk/gsdk.cpp

```diff
@@ -403,81 +472,144 @@ namespace Microsoft
                 decodeHeartbeatResponse(m_receivedData);
             }
 
-            Microsoft::Azure::Gaming::GSDKInternal& GSDKInternal::get()
+            Microsoft::Azure::Gaming::GSDKInternal* GSDKInternal::get()
             {
-                std::unique_lock<std::mutex> lock(m_gsdkInitMutex);
```

Comment:
Moving the instantiation of m_instance to start() is great. While it's admittedly weird to call start multiple times, I think it's still legitimate that start could/should retain the mutex lock around the instantiation of m_instance to prevent multiple simultaneous instantiation. Based on your issue description, a mutex in start() won't affect the get()'s that are called during shutdown.